crawler.c

```diff
@@ -604,6 +655,12 @@ int lru_crawler_start(uint8_t *ids, uint32_t remaining,
         return -1;
     }
 
+    /* hash table walk only supported with metadump for now. */
+    if (type != CRAWLER_METADUMP && ids == NULL) {
```

Comment:
nit: We can catch this much earlier at lru_crawler_crawl() at L726 without needing to take the lock.

Reply:
lru_crawler_crawl() is "externally" facing, so future-proofing puts this check a bit deeper, though it ends up giving a non user friendly error.